src/test/java/com/commercetools/config/ctpTypes/AggregatedCtpTypesValidationResultTest.java

```diff
@@ -0,0 +1,167 @@
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class AggregatedCtpTypesValidationResultTest {
+
+    @Test
+    public void ofEmpty() throws Exception {
+        AggregatedCtpTypesValidationResult result = AggregatedCtpTypesValidationResult.ofEmpty();
+        assertThat(result).isNotNull();
+        assertThat(result.isEmpty()).isTrue();
+        assertThat(result.hasErrorMessage()).isFalse();
+        assertThat(result.getAggregatedErrorMessage()).isEmpty();
+        assertThat(result.hasUpdatedTypes()).isFalse();
+        assertThat(result.getUpdatedTypes()).isEmpty();
+    }
+
+    @Test
+    public void ofErrorMessage() throws Exception {
+        assertThatIllegalArgumentException().isThrownBy(() -> AggregatedCtpTypesValidationResult.ofErrorMessage(""));
+        assertThatIllegalArgumentException().isThrownBy(() -> AggregatedCtpTypesValidationResult.ofErrorMessage("   "));
+
+        AggregatedCtpTypesValidationResult result = AggregatedCtpTypesValidationResult.ofErrorMessage("It's amazing how much \"mature wisdom\" resembles being too tired");
+        assertThat(result).isNotNull();
+        assertThat(result.isEmpty()).isFalse();
+        assertThat(result.hasErrorMessage()).isTrue();
+        assertThat(result.getAggregatedErrorMessage()).isEqualTo("It's amazing how much \"mature wisdom\" resembles being too tired");
+        assertThat(result.hasUpdatedTypes()).isFalse();
+        assertThat(result.getUpdatedTypes()).isEmpty();
+    }
+
+    @Test
+    public void ofUpdatedTypes() throws Exception {
+        assertThatIllegalArgumentException().isThrownBy(() -> AggregatedCtpTypesValidationResult.ofUpdatedTypes(emptyList()));
+
+        AggregatedCtpTypesValidationResult result = AggregatedCtpTypesValidationResult.ofUpdatedTypes(asList(
+                Pair.of("tenant1", asList(mockType("t1_key1"), mockType("t1_key2"))),
+                Pair.of("tenant2", singletonList(mockType("t2_key1")))
+        ));
+        assertThat(result).isNotNull();
+        assertThat(result.isEmpty()).isFalse();
+        assertThat(result.hasErrorMessage()).isFalse();
+        assertThat(result.getAggregatedErrorMessage()).isEmpty();
+        assertThat(result.hasUpdatedTypes()).isTrue();
+
+        String concatTenantNamesTypeKeys = concatUpdatedTypesToString(result);
+        assertThat(concatTenantNamesTypeKeys).isEqualTo("tenant1#t1_key1,t1_key2;tenant2#t2_key1");
+    }
+
+    @Test
+    public void executeAndAggregateTenantValidationResults_withEmptyActions_shouldReturnEmptyResult() throws Exception {
```

Comment:
Why do we need these `throws Exception` declarations when we don't need them (i.e. when no exception will be thrown in the test)?

Reply:
it's default Intellij IDEA autogenerate "feature". I just ignore it.